internal/txprocessor/txprocessor.go

```diff
@@ -69,7 +77,26 @@ func (r TXProcessor) GetSubmitNotificationChannel() <-chan relay.PendingSubmitte
 func (r TXProcessor) submitTxWithProofs(queryID uint64, block *neutrontypes.Block) error {
 	proofStart := time.Now()
 	hash := hex.EncodeToString(tmtypes.Tx(block.Tx.Data).Hash())
-	neutronTxHash, err := r.submitter.SubmitTxProof(queryID, block)
+	var neutronTxHash string
+	err := retry.Do(func() error {
+		var err error
+		r.logger.Info("going to try submit tx prof", zap.Uint64("query_id", queryID))
+		neutronTxHash, err = r.submitter.SubmitTxProof(queryID, block)
+		if err != nil {
+			r.logger.Debug("tx submit error", zap.Error(err))
+		}
+		return err
+	},
+		retry.RetryIf(func(err error) bool {
+			return strings.Contains(err.Error(), "exceeds gas limit") ||
+				strings.Contains(err.Error(), "error making abci query for gas calculation") ||
+				strings.Contains(err.Error(), "error broadcasting sync transaction")
```

Comment:
So, we are trying to resubmit tx if there are some problems with gas (first two conditions). But which error does the third one describe?

Reply:
Will the transaction be resubmitted if an error happens during `Sudo` callback?

Reply:
Good Q. As for now - no. But IMO we can add this case as recoverable error. What do you think?

Reply:
I think we shouldn't resubmit txs in this case, yeah

Reply:
Also in case of gas problems you resubmit txs with the same parameters (gas-prices, gas amount, etc). What's the point? Like, won't you get the same error over and over again, since you trying to submit tx with the same gas amount?

Reply:
The state of contract can be changed in time and it can affect gas consumption, also contract may be redeployed

Reply:
> I think we shouldn't resubmit txs in this case, yeah

why? Error may depend on contract state, also contract may be redeployed as well